### ISSUE TITLE
8$: error on '\u{63}onstructor' in TypeScript

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#1003]
The following code is legal in JavaScript but is illegal in TypeScript:
```javascript
class C {
  \u{63}onstructor() {}  // equivalent to: constructor() {}
}
```
We should either report this issue to upstream TypeScript or make quick-lint-js diagnose this code is illegal TypeScript code.

**Comment by jaitjacob** [comment]
I reported this error to upstream TS which can be found [here](https://github.com/microsoft/TypeScript/issues/54196). 

I'd like to contribute a fix to this issue would I be touching test/test-parse-expression.cpp & src/quick-lint-js/fe/parse-expression.cpp? are there other files I need to be looking at?

**Comment by strager** [comment]
> test/test-parse-expression.cpp (create a test first) & src/quick-lint-js/fe/parse-expression.cpp(code against test)?

I think test/test-parse-typescript-class.cpp and src/quick-lint-js/fe/parse-class.cpp are probably what you'd touch instead.

> are there other files I need to be looking at?

To create a new diagnostic, see this documentation: https://quick-lint-js.com/contribute/create-diagnostic/

**Comment by jaitjacob** [comment]
the diag report i added was based on my assumption that `parse-class.cpp` guarantees that code within
[`case token_type::reserved_keyword_with_escape_sequence:{...}`](https://github.com/quick-lint/quick-lint-js/blob/d20bb8693493009fe8780adf5ca0e5a46134b9ba/src/quick-lint-js/fe/parse-class.cpp#L420)  would never execute if current keyword does not contain an escape sequence. this assumption turned out to be false. so I've reverted to my failing test commit for a fresh start.

**what next:** as suggested by Strager on the stream I could check for `escape_sequence_list* identifier_escape_sequences` in the `token` type returned by `peek()` to trigger the diag.

**Comment by jaitjacob** [comment]
the diag report i added was based on my assumption that `parse-class.cpp` guarantees that code within
[`case token_type::reserved_keyword_with_escape_sequence:{...}`](https://github.com/quick-lint/quick-lint-js/blob/d20bb8693493009fe8780adf5ca0e5a46134b9ba/src/quick-lint-js/fe/parse-class.cpp#L420)  would never execute if current keyword does not contain an escape sequence. this assumption turned out to be false. so I've reverted to my failing test commit for a fresh start.

**what next:** as suggested by Strager on the stream I could check for `escape_sequence_list* identifier_escape_sequences` in the `token` type returned by `peek()` to trigger the diag.
